src/main/java/io/cdap/plugin/google/sink/GoogleDriveSinkConfig.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.google.sink;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.annotation.Description;
+import io.cdap.cdap.api.annotation.Macro;
+import io.cdap.cdap.api.annotation.Name;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.FailureCollector;
+import io.cdap.plugin.google.common.GoogleDriveBaseConfig;
+
+import javax.annotation.Nullable;
+
+/**
+ * Configurations for Google Drive Batch Sink plugin.
+ */
+public class GoogleDriveSinkConfig extends GoogleDriveBaseConfig {
+  public static final String SCHEMA_NAME_FIELD_NAME = "schemaNameFieldName";
+  public static final String SCHEMA_BODY_FIELD_NAME = "schemaBodyFieldName";
+
+  @Nullable
+  @Name(SCHEMA_NAME_FIELD_NAME)
+  @Description("Name of the schema field which will be used as name of file.")
+  @Macro
+  protected String schemaNameFieldName;
+
+  @Name(SCHEMA_BODY_FIELD_NAME)
+  @Description("Name of the schema field which will be used as body of file.")
+  @Macro
+  protected String schemaBodyFieldName;
+
+  public void validate(FailureCollector collector, Schema schema) {
+    super.validate(collector);
+
+    // validate body field is in schema and has valid format
+    validateSchemaField(collector, schema, SCHEMA_BODY_FIELD_NAME, schemaBodyFieldName, Schema.of(Schema.Type.BYTES));
+
+    // validate name field is in schema and has valid format
+    validateSchemaField(collector, schema, SCHEMA_NAME_FIELD_NAME, schemaNameFieldName, Schema.of(Schema.Type.STRING));
+  }
+
+  private void validateSchemaField(FailureCollector collector, Schema schema, String propertyName,
+                                   String propertyValue, Schema requiredSchema) {
```

Comment:
You have to check if field is not logical type and if it's nullable than to get non-nullable part.
Code example from one of my plugins:
```
    Schema.Field vertexField = inputSchema.getField(vertexType);
    if (vertexField == null) {
      failureCollector.addFailure(String.format("Field '%s' is not present in input schema.", vertexType),
                                  null).withConfigProperty(VERTEX)
        .withInputSchemaField(VERTEX, null);
    } else {
      Schema vertexFieldSchema = vertexField.getSchema();

      if (vertexFieldSchema.isNullable()) {
        vertexFieldSchema = vertexFieldSchema.getNonNullable();
      }

      if (vertexFieldSchema.getLogicalType() != null || vertexFieldSchema.getType() != Schema.Type.STRING) {
        failureCollector.addFailure(String.format("Field '%s' must be of type 'string' but is of type '%s'.",
                                                  vertexField.getName(), vertexFieldSchema.getDisplayName()),
                                    null).withConfigProperty(VERTEX)
          .withInputSchemaField(VERTEX, null);
      }
    }
```

Also use .withInputSchemaField